esrally/storage/_manager.py

```diff
@@ -15,128 +15,109 @@
+        :param cfg: Configuration object.
         :param client: _client.Client instance used to allocate/reuse storage adapters.
-        :param multipart_size: length of every part when working with multipart.
-        :param max_workers: max number of connections per remote server when working with multipart.
+        :param executor: Executor instance used to execute transfer operations.
         """
+        self.cfg = cfg
         self._client = client
         self._executor = executor
         self._lock = threading.Lock()
 
-        local_dir = os.path.expanduser(local_dir)
+        local_dir = os.path.expanduser(cfg.local_dir)
         if not os.path.isdir(local_dir):
             os.makedirs(local_dir)
-
         self._local_dir = local_dir
-        if monitor_interval <= 0:
-            raise ValueError(f"invalid monitor interval: {monitor_interval} <= 0")
 
-        self._transfers: list[Transfer] = []
+        self._transfers: dict[str, Transfer] = {}
 
-        if max_workers < 1:
-            raise ValueError(f"invalid max_workers: {max_workers} < 1")
-        self._max_workers = max_workers
+        if cfg.max_workers < 1:
+            raise ValueError(f"invalid max_workers: {cfg.max_workers} < 1")
 
-        if max_connections < 1:
-            raise ValueError(f"invalid max_connections: {max_connections} < 1")
-        self._max_connections = max_connections
+        if cfg.max_connections < 1:
+            raise ValueError(f"invalid max_connections: {cfg.max_connections} < 1")
 
-        if multipart_size < 1024 * 1024:
-            raise ValueError(f"invalid multipart_size: {multipart_size} < {1024 * 1024}")
-        self._multipart_size = multipart_size
+        if cfg.multipart_size < 1024 * 1024:
+            raise ValueError(f"invalid multipart_size: {cfg.multipart_size} < {1024 * 1024}")
 
-        self._monitor_timer = ContinuousTimer(interval=monitor_interval, function=self.monitor, name="esrally.storage.transfer-monitor")
+        if cfg.monitor_interval <= 0:
+            raise ValueError(f"invalid monitor interval: {cfg.monitor_interval} <= 0")
+        self._monitor_timer = ContinuousTimer(interval=cfg.monitor_interval, function=self.monitor, name="esrally.storage.transfer-monitor")
         self._monitor_timer.start()
 
     def shutdown(self):
         with self._lock:
             transfers = self._transfers
-            self._transfers = []
-        for tr in transfers:
-            tr.close()
+            self._transfers = {}
+        LOG.info("Shutting down transfer manager...")
+        for tr in transfers.values():
+            try:
+                if tr.finished:
+                    continue
+                LOG.warning("Cancelling transfer: %s...", tr)
+                tr.close()
+            except Exception as ex:
+                LOG.error("error closing transfer: %s, %s", tr.url, ex)
         self._monitor_timer.cancel()
+        LOG.info("Project manager shut down.")
```

Comment:
```suggestion
        LOG.info("Transfer manager shut down.")
```

BTW, I do not see this log in `rally.log`. It does appear in Thespian logs, but apparently logger actor gets shutdown before it manages to process that?